server/reload.go

```diff
@@ -293,6 +293,15 @@ func (a *authTimeoutOption) Apply(server *Server) {
 	server.Noticef("Reloaded: authorization timeout = %v", a.newValue)
 }
 
+// tagsOption implements the option interface for the `tags` setting.
+type tagsOption struct {
+	noopOption // Not authOption because this is a no-op; will be reloaded with options.
+}
+
+func (u *tagsOption) Apply(server *Server) {
+	server.Noticef("Reloaded: tags")
```

Comment:
I like this addition, but please note, AFAIK we have never written updates to config files as a result of a $SYS msg. That was by design. So we should carefully consider this and make sure we are committed here. Also need to loop in Phil imo.

There may be a better way to achieve the same goal of temporarily notifying the metaleader not to consider this server for placement, etc.

Reply:
This is not triggered by an update on a $SYS message. 
This is driven by configuration only.

So if, for or whatever reason, we need placement to stop for a server. just place the new tag and reload the config.
Separating move and stopping of placement into two features was an intentional decision.

Enabling stopping of placement via server tags also piggy backs onto an existing mechanism (propagation already implemented) and is robust against restarts. (persistence is done via config) (just consider, meta leader changing, actual server being restarted for whatever reason...)
Furthermore, reload of tags is a worthwhile feature in its own right. 
(With this, tags can also convey the config version)

Reply:
ok, I think black box operations are desirable, but don't think we should be updating the server config. Sounds like we should discuss before more progress is made.

Reply:
sent out an invite for later today

Reply:
Thanks, was chatting with Caleb as well this am, should be a short call I think.

Reply:
I think there is usecase for both:

 * Normal operations we need to be able to tell a machine to stop accepting stuff and evict stuff without config changes, it should be soft and in memory only driven by a $SYS style query. It should not exit once all is evicted - its not a LDM style operation
 * A config override so that, while preparing a new machine or dealing with a machine that is known to have some issue like disk performance, we can stop accepting assets but have it up for monitoring etc but it could still be fine for core traffic

Related, we need a $SYS style way to signal a config reload.

Reply:
If we don't have a way to deliver new configs via NATS and $SYS not sure that is useful? Just have the extra layer that delivers the new config also signal after receipt and successful check?

Reply:
No, not new configs, just signal reloads that responds back with success/err and reason.

For placement candidate it could be something that persists into the raft layer and resets either using API or force flag in the config file. But probably fine just to be in memory while the machine is up for normal operation needs.

Wdyt @davidkemper

Reply:
It is my understanding that putting this setting into the metaleader would open any number of cans of worms.

Where I'm going to make a big disruptive change recreating a vm, IMHO it's an acceptable compromise to set some persistent server tag as a prep step.

The prime use case is keeping r=1 highly available during a disruptive change. In this case I can spin up a replacement in the same availability zone as a target to keep all data in the same data center.

Arguably a simple tag on the doomed server would be very robust and intentional even across multiple failure modes, including having to upgrade the servers halfway through...

Reply:
We can always add a dedicated system call with the necessary bells and whistles later. 
The config based approach gives us exactly what we want for very little effort.

Reply:
I wonder if we set the tag on a server that is also the meta leader, its steps down first, delays until new leader ok then sends tag metadata about placement?